test-tools/wamr-ide/VSCode-Extension/src/utilities/dockerUtilities.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+import * as vscode from 'vscode';
+import * as cp from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs';
+import { getWAMRExtensionVersion } from './lldbUtilities';
+import { downloadFile, unzipFile } from './directoryUtilities';
+import { SelectionOfPrompt, Status } from '../constants';
+
+const DOCKER_IMAGES_TEM_FOLDER_NAME = 'docker-resource';
+
+const execShell = (cmd: string) =>
+    new Promise<string>((resolve, reject) => {
+        cp.exec(cmd, (error, result) => {
+            if (error) {
+                return reject(error);
+            }
+            return resolve(result);
+        });
+    });
+
+export async function promptSetupDockerImages(
+    context: vscode.ExtensionContext
+): Promise<string> {
```

Comment:
Looks like this function returns `SelectionOfPrompt` or `Status`, but not string. You might consider having a single type being returned (e.g. `Status`) and extend it with `Skip` or `Cancel` field. Also, I wonder why typechecker didn't capture this - is enum treated as string in TypeScript or is the typechecker disabled?